x-pack/plugins/fleet/server/services/preconfiguration.ts

```diff
@@ -35,13 +35,84 @@ import { agentPolicyService, addPackageToAgentPolicy } from './agent_policy';
 import type { InputsOverride } from './package_policy';
 import { overridePackageInputs } from './package_policy';
 import { appContextService } from './app_context';
+import { outputService } from './output';
 
 interface PreconfigurationResult {
   policies: Array<{ id: string; updated_at: string }>;
   packages: string[];
   nonFatalErrors: PreconfigurationError[];
 }
 
+function isPreconfiguredOutputDifferentFromCurrent(
+  existingOutput: Output,
+  preconfiguredOutput: Partial<Output>
+): boolean {
+  return (
+    existingOutput.is_default !== preconfiguredOutput.is_default ||
+    existingOutput.name !== preconfiguredOutput.name ||
+    existingOutput.type !== preconfiguredOutput.type ||
+    (preconfiguredOutput.hosts &&
+      !isEqual(
+        existingOutput.hosts?.map(normalizeHostsForAgents),
+        preconfiguredOutput.hosts.map(normalizeHostsForAgents)
+      )) ||
+    existingOutput.ca_sha256 !== preconfiguredOutput.ca_sha256 ||
+    existingOutput.config_yaml !== preconfiguredOutput.config_yaml
+  );
+}
+
+export async function ensurePreconfiguredOutputs(
+  soClient: SavedObjectsClientContract,
+  outputs: PreconfiguredOutput[]
+) {
+  await Promise.all(
+    outputs.map(async (output) => {
+      const existingOutput = await outputService.get(soClient, output.id).catch((err) => {
```

Comment:
Could we get all of the saved objects in one call instead of one-by-one?

Reply:
👍 Should be able to use `bulkGet` instead I think